src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/PackageSourceTelemetry.cs

```diff
@@ -0,0 +1,403 @@
+                internal static readonly string Retries = "metadata.retries";
+                internal static readonly string Cancelled = "metadata.cancelled";
+                internal static readonly string Failed = "metadata.failed";
+
+                internal static class Bytes
+                {
+                    internal static readonly string Total = "metadata.bytes.total";
+                    internal static readonly string Max = "metadata.bytes.max";
+                }
+
+                internal static class Http
+                {
+                    internal static readonly string StatusCodes = "metadata.http.statuscodes";
+                }
+
+                internal static class Timing
+                {
+                    internal static readonly string Min = "metadata.timing.min";
+                    internal static readonly string Mean = "metadata.timing.mean";
+                    internal static readonly string Max = "metadata.timing.max";
+                }
+
+                internal static class Header
+                {
+                    internal static class Timing
+                    {
+                        internal static readonly string Min = "metadata.header.timing.min";
+                        internal static readonly string Mean = "metadata.header.timing.mean";
+                        internal static readonly string Max = "metadata.header.timing.max";
+                    }
+                }
+            }
+
+            internal static class Nupkg
+            {
+                internal static readonly string Requests = "nupkg.requests";
+                internal static readonly string Successful = "nupkg.successful";
+                internal static readonly string Retries = "nupkg.retries";
+                internal static readonly string Cancelled = "nupkg.cancelled";
+                internal static readonly string Failed = "nupkg.failed";
+
+                internal static class Bytes
+                {
+                    internal static readonly string Total = "nupkg.bytes.total";
+                    internal static readonly string Max = "nupkg.bytes.max";
+                }
+
+                internal static class Http
+                {
+                    internal static readonly string StatusCodes = "nupkg.http.statuscodes";
+                }
+
+                internal static class Timing
+                {
+                    internal static readonly string Min = "nupkg.timing.min";
+                    internal static readonly string Mean = "nupkg.timing.mean";
+                    internal static readonly string Max = "nupkg.timing.max";
+                }
+
+                internal static class Header
+                {
+                    internal static class Timing
+                    {
+                        internal static readonly string Min = "nupkg.header.timing.min";
+                        internal static readonly string Mean = "nupkg.header.timing.mean";
```

Comment:
I think the median is more useful than the mean.

Reply:
Specifically, the distribution matters more, so std deviation probably tells us more.

Reply:
I agree that median is more useful, but that would require us to store every duration, so that we can sort and then select the middle one. I don't remember the formula for standard deviation, but I think it also needs every data point to be stored. We can consider it for a future PR, but I don't feel comfortable doing this now.

Reply:
ok, create an issue for that please.

Reply:
I suggest the following changes:

* Remove mean and don't add median.  You could also remove min and max, but it's also OK if you want to leave them in.
* Add totals at each package source level
* Add totals for all package sources at the restore summary level

For each package source diagnostics event log the total (sum) for each of time, bytes, and requests.  Metadata/nupkg and header/content values would be included in the sums.  Example:
```C#
telemetry[PropertyNames.Timing.Total] = /* sum of all metadata and nupkg times for all package sources in milliseconds */
telemetry[PropertyNames.Bytes.Total] = /* sum of all bytes transferred */
telemetry[PropertyNames.Requests.Total] = /* count of all requests */
```
Additionally, add the total (sum) for each time, bytes, and requests for **all** packages sources on the restore summary event plus processor count (`Environment.ProcessorCount`).